### DISCUSSION TITLE
Resilience and outage time and date

### DISCUSSION THREAD

**Discussion by NLaws** [NREL/REopt-API-Analysis#60] [4 - Resilience Questions] [answered]
See https://github.com/NREL/REopt-API-Analysis/issues/59 for a discussion on Resilience scenarios, the `resilience_check_flag`, and `outage_start_time_step`

**Comment by NLaws** [comment, marked as answer]
A summary:

Resilience scenarios in the web app are the same as adding an outage to the model.  When an outage is included in the model the critical load must be met during the outage. (The web app also runs the same scenario without the outage for the "Financial" comparison on the results page.)

The `outage_start_hour` is an old input that has been kept for backwards compatibility. You can use the `outage_start_time_step` value for 15 minute interval load data. 

Our wiki has a link to all of the API inputs:
- the wiki: https://github.com/NREL/REopt-API-Analysis/wiki/Job-Inputs
- API inputs: https://github.com/NREL/REopt_Lite_API/blob/master/reo/nested_inputs.py
- shortcut to `outage_start_time_step` input: https://github.com/NREL/REopt_API/blob/fbc70f3b0cdeec9ee220266d6b3b0c5d64f257a6/reo/nested_inputs.py#L588

`resilience_check_flag` is not an input; It is true if the critical load can be met during the outage with the existing generation technologies:
https://github.com/NREL/REopt_API/blob/bb85165ecfad7ff478e80b7746a8a755f658167d/reo/src/load_profile.py#L769
In the next major API version the `resilience_check_flag` output will be renamed to `bau_critical_load_met`:
https://github.com/NREL/REopt_API/blob/fb89f2c17fba2aec8d0d5d1eaa901415dfd129b0/job/models.py#L799
(`bau` = business-as-usual)

The `resilience_check_flag` only applies to the business-as-usual scenario. For every "Scenario" that you POST to the job endpoint two scenarios are run:
1. The business-as-usual (BAU) scenario, which determines the cost of meeting the (electric and thermal) loads with the grid and any _existing_ technologies; and
2. The optimal scenario, which determines the cost-optimal size and dispatch of considered technologies.

A false `resilience_check_flag` indicates that the critical load is not entirely met in the BAU scenario (if you have no existing technologies then zero percent of the critical load is met in the BAU scenario).  If you want to include an existing Generator for example then you must provide the generator capacity in the `existing_kw` input field.

One reason for differences in webtool and API scenarios could be that the webtool only runs hourly models (even if 15 minute data is provided). We do this in the webtool because 15 minute scenarios typically take longer to solve than hourly scenarios.

**Reply by Janeyyt** [reply]
Thank you!
